src/core/common/clearable.hpp

```diff
@@ -53,7 +55,7 @@ template <typename Type> class Clearable
 {
 public:
     void Clear(void) { memset(reinterpret_cast<void *>(this), 0, sizeof(Type)); }
-};
+} OT_TOOL_PACKED_END;
```

Comment:
Is there any build option provided by Keil toolchain that allows user to relax the requirement of packed base-classes (specifically in the cases where base-class does not define any member variable).

If not, I would say the the approach to define these as packed only under Keil toolchain is the best approch among 3 options you mentioned.

Is there an `#ifdef` we can add here to limit this line to only Keil tool chain?

Reply:
Unfortunately Keil compiler documentation does not specify such option.

Reply:
But there is the compile time check that you can use, see [here](https://github.com/openthread/openthread/blob/master/examples/platforms/nrf528xx/src/flash.c#L72).

```
#if defined(__CC_ARM)
```

Reply:
@LuDuda I used it, see the commit https://github.com/openthread/openthread/commit/02af093fa69ddb3bd8dca510d56d6df8a9845b84. This solution passes all build tests (GCC, IAR, KEIL).